walnut/converter/converter.go

```diff
@@ -138,9 +138,9 @@ func ExtractTaskClasses(dplDump Dump, envModules []string) (tasks []*task.Class,
 func sanitizeCmdLineArgs (input []string, taskName string) (output []string) {
 	for index, value := range input {
 		// Check args for dump arguments and remove them
-		if  strings.Contains(value, "--dump-workflow") ||
-			strings.Contains(value, "--dump-workflow-file") ||
-			strings.Contains(value, ".json") && input[index-1] == "--dump-workflow-file" {} else {
+		if  !strings.Contains(value, "--dump-workflow") ||
+			!strings.Contains(value, "--dump-workflow-file") ||
+			!strings.Contains(value, ".json") && input[index-1] != "--dump-workflow-file" {
```

Comment:
Let's unpack this: you have 3 clauses separated by `||`, and the last of these has a `&&` inside (`&&` is evaluated before `||` similarly to how multiplication takes precedence over addition). You want to negate the whole if-expression to get rid of the unelegant `{} else` construct, but much like in algebra you cannot simply negate each of the operands to negate the whole clause. In other words, it is **not true** that NOT(A OR B) == NOT(A) OR NOT(B).

You could always write
```
if !(strings.Contains(value, "--dump-workflow") ||
	(value, "--dump-workflow-file") ||
	(value, ".json") && input[index-1] == "--dump-workflow-file") {
```
and this solution would be correct, but if you wish to make the expression more elegant and only do negations on the innermost statements, you need to apply De Morgan's laws.

I advise you take a look at https://en.wikipedia.org/wiki/De_Morgan%27s_laws and (protip) remember them, because negating a clause is a very common task that pops up time and again in computer science.

Reply:
Understood. Apologies for the oversight. I ended up with:
```go
        if !(strings.Contains(value, "--dump-workflow") ||
            (strings.Contains(value, ".json") && input[index-1] == "--dump-workflow-file")) {
```

After some deliberation, I noticed we need not check for `--dump-workflow-file` as that should be handled by the `--dump-workflow` check. I also parenthesized the second check to make sure the `&&` executes when we need it to.